common/input_file.py

```diff
@@ -7,5 +7,18 @@ def read_numbers(input_filename):
     """
         Read a file and return a list of numbers
     """
+    return get_transformed_input(input_filename, int)
+
+def read_strings(input_filename):
+    """
+        Read a file and return a list of strings
+    """
+    return get_transformed_input(input_filename, str)
+
+def get_transformed_input(input_filename, transform_function):
+    """
+        Read all the lines of a file and transform it according to
+        transform_function
+    """
     with open(input_filename) as input_file:
-        return [int(n) for n in input_file.readlines()]
+        return [transform_function(line).strip() for line in input_file.readlines()]
```

Comment:
read_numbers should fail now. You should call strip() on line, not the result of the transform, OR you could pass in a lambda in read_strings to do the strip after calling str.

Reply:
I think I always want the \n stripped off